MapboxNavigation/NavigationViewController.swift

```diff
@@ -293,6 +293,31 @@ public class NavigationViewController: NavigationPulleyViewController, RouteMapV
             return
         }
         
+        // If the user is in a parking lot, don't reroute
+        guard let mapView = mapView else { return }
+        guard let style = mapView.style else { return }
+        
+        let roadLabelLayerIdentifier = "parkingLot"
+        var streetsSources = style.sources.flatMap { $0 as? MGLVectorSource }.filter { $0.isMapboxStreets}
+        
+        if streetsSources.isEmpty {
+            let source = MGLVectorSource(identifier: "mapboxStreetsv7", configurationURL: URL(string: "mapbox://mapbox.mapbox-streets-v7")!)
+            style.addSource(source)
+            streetsSources.append(source)
```

Comment:
This logic is duplicated from the street label code. Would it be possible to factor it out into an extension method on MGLStyle?